Source/WebKit/_WebKit_SwiftUI/API/WebView.swift

```diff
@@ -36,13 +36,46 @@ public struct WebView: View {
     ///
     /// - Parameter page: The ``WebPage`` that should be associated with this ``WebView``. It is a programming error to create multiple ``WebView``s with the same ``WebPage``.
     public init(_ page: WebPage) {
-        self.page = page
+        self.storage = .webPage(page)
     }
 
-    private let page: WebPage
+    /// Create a new WebView with the specified URL.
+    ///
+    /// For example, you can create a WebView that displays one of two URLs depending on the state of a toggle:
+    ///
+    ///     struct URLView: View {
+    ///         @State private var url: URL? = nil
+    ///         @State private var toggle = false
+    ///
+    ///         var body: some View {
+    ///             VStack {
+    ///                 Button("Toggle") {
+    ///                     toggle.toggle()
+    ///                 }
+    ///                 WebView(url: url)
+    ///             }
+    ///             .onChange(of: toggle, initial: true) {
+    ///                 url = toggle ? URL(string: "https://www.webkit.org") : URL(string: "https://www.apple.com")
+    ///             }
+    ///         }
+    ///     }
+    ///
+    /// - Parameter url: The URL to display in the view. If this value is non-nil or changes to become a non-nil value, the new URL is loaded into the view.
+    public init(url: URL?) {
+        self.storage = .state(State(initialValue: WebPage()), url)
+    }
+
+    private let storage: Storage
 
     public var body: some View {
-        WebViewRepresentable(page: page)
+        WebViewRepresentable(page: storage.webPage)
+            .onChange(of: storage.url, initial: true) {
+                guard let url = storage.url else {
+                    return
+                }
+
+                storage.webPage.load(URLRequest(url: url))
```

Comment:
Like the lack of control over this part of the behavior is what worries me that this API may become overly attractive, but then still require refactoring down the line.